### ISSUE TITLE
[1.17] pcall/custom event error due to no stations in schedule

### ISSUE THREAD

**Issue by doktorstick** [Choumiko/SmartTrains#43]
I have two trains using the same schedule. I updated one train and saved the schedule. The other train did not update. In fact, it was stuck with a "no path", which was the cause for my edit to the schedule. I deleted the offending station hoping that it would resync its schedule at the next station; it did not. Instead it reverted to the broken schedule.

What follows is what I did to break SmartTrains.

1) I deleted all of stations from the broken train. I *think* I removed the active checkbox, too.
2) I closed the train window. An error was emitted when this happened, but the subsequent error scrolled this one off the screen.
3) I opened the train window. Smart Trains showed that it had an empty schedule.
4) I checked "active" for the schedule I wanted the new train to inherit.

At this point, SmartTrains is stuck in a loop squawking at me. I'm typing this from the screen; pardon any typos:
```
Error while running event Custom event (ID 74) __SmartTrains__/Train.lua:638:attempt to get length of field 'records' (a nil value) stack traceback: __SmartTrains__/control.lua:1210: in function <__SmartTrains__/control.lua:1202> [C]: in function 'pcall' __SmartTrains__/control.lua:1202: in function <__SmartTrains__/control.lua:1111>
```

If I open the train window, the errors stop, but there is no SmartTrains companion window. Adding at least one station and closing the train window returned functionality and I was able to get it to accept my "active" schedule.

**Comment by Choumiko** [comment]
The error should be fixed in the latest commit 3b3b8a10ae61d33e646b08174f1b6a9ab6f132c4

About the broken schedule: Did the train update to the correct schedule and just couldn't get there or did SmartTrains update to a wrong schedule?

**Comment by doktorstick** [comment]
It updated it to the wrong (old) schedule. If I had to wager, my guess it
was two trains shared the same schedule. I made the modification and  sent
it to the new station (thereby engaging auto mode). At some point along its
path--probably after the first station it arrived--it reverted to the old
schedule. I wasn't paying strict attention.
